src/wallet/wallet.h

```diff
@@ -635,7 +635,30 @@ class CWallet final : public WalletStorage, public interfaces::Chain::Notificati
 
     std::optional<int64_t> GetOldestKeyPoolTime() const;
 
-    std::set<CTxDestination> GetLabelAddresses(const std::string& label) const EXCLUSIVE_LOCKS_REQUIRED(cs_wallet);
+    // Filter struct for 'ListAddrBookAddresses'
+    struct AddrBookFilter {
+        // Fetch addresses with the provided label
+        std::optional<std::string> m_op_label{std::nullopt};
+        // Always filter change addresses by default
+        bool filter_change{true};
```

Comment:
nit: The usage of the verb "filter" is potentially confusing here, as it's not clear whether it means inclusion or exclusion; if the filter structure has `m_op_label` set, it means to only end up with addresses of that label, but if `filter_change` is true, all change addresses are ignored (i.e. the opposite logic). Probably use something like

```suggestion
        // Never include change addresses by default
        bool ignore_change{true};
```

Reply:
sure, updated.